lib/iris/fileformats/netcdf.py

```diff
@@ -1190,9 +1190,9 @@ def write(
 
         # Ensure that attributes are CF compliant and if possible to make them
         # compliant.
-        self.check_attribute_compliance(cube, cube.lazy_data())
+        self.check_attribute_compliance(cube, cube.dtype)
         for coord in cube.coords():
-            self.check_attribute_compliance(coord, coord.points)
```

Comment:
This was the problem line that realised all the points
(though bounds were ok, interestingly)

I think we **_could_** just replace the `points` here with `core_points()`
However, I also realised that `check_attribute_compliance` call doesn't actually _use_ the data array at all, it only uses the _dtype_ of it.
So, I've replaced that throughout, as an alternative to passing a 'safe' version of the data in the problem location.
Unfortunately that does causes a lot of noisy changes in the tests (though very little in functional code).